src/wavm.cpp

```diff
@@ -185,24 +185,29 @@ ExecutionResult WavmEngine::execute(
     ensureCondition(false, ContractValidationFailure, "Bug in wavm: didn't check bounds before allocation");
   }
 
-  // next set up the host module.
-  // Note: in ewasm, we create a new VM for each call to a module, so we must instantiate a new host module for each of these VMs, this is inefficient, but OK for prototyping.
+  // compile the module (not sure what this does)
+  Runtime::Module* module = Runtime::compileModule(irmodule);
```

Comment:
If `irmodule` is not used after this, then I think we should create a helper so that `IR::Module` is freed earlier to reduce memory usage.

Reply:
Yes, sounds like a good design. I was hoping for simplicity until it is functional, then do optimizations like this.

Reply:
In general memory usage may also affect benchmark results.